spec/keycard/request/cosign_attributes_spec.rb

```diff
@@ -3,14 +3,15 @@
 require "keycard/request/cosign_attributes"
 
 RSpec.describe Keycard::Request::CosignAttributes do
+  let(:attributes) { described_class.new(rack_request) }
+
   context "with typical forwarded headers" do
-    let(:request) do
-      double('request', env: {
+    let(:rack_request) do
+      double(:rack_request, env: {
```

Comment:
No objection, but I'd like to understand what `rack_request` communicates to you that `request` does not, and if we have a different place to do that and keep this more concise and meanwhile less specific.

Reply:
Clarification: I personally think that include "rack" is extraneous detail -- it's an HTTP request with an env property, whatever layering or adapter is applied.

Reply:
It clearly communicates that the double is not shadowing an object within keycard via referencing the most common case / the design target.